### ISSUE TITLE
plot_resid() with bokeh

### ISSUE THREAD

**Issue by DougBurke** [sherpa/sherpa#1991]
I get something like the following after trying to show the output of `plot_resid()` with bokeh:

```
File <string>:1, in plot_resid(id, replot, overplot, clearwindow, **kwargs)

File /lagado.real/sherpa/sherpa-main/sherpa/ui/utils.py:14370, in Session.plot_resid(self, id, replot, overplot, clearwindow, **kwargs)
  14293 """Plot the residuals (data - model) for a data set.
  14294 
  14295 This function displays the residuals (data - model) for a data
   (...)
  14366 
  14367 """
  14369 plotobj = self.get_resid_plot(id, recalc=not replot)
> 14370 self._plot(plotobj, overplot=overplot, clearwindow=clearwindow,
  14371            **kwargs)

File /lagado.real/sherpa/sherpa-main/sherpa/ui/utils.py:13321, in Session._plot(self, plotobj, **kwargs)
  13305 """Display a plot object
  13306 
  13307 Parameters
   (...)
  13317 
  13318 """
  13320 with sherpa.plot.backend:
> 13321     plotobj.plot(**kwargs)

File /lagado.real/sherpa/sherpa-main/sherpa/plot/__init__.py:2325, in ResidPlot.plot(self, overplot, clearwindow, **kwargs)
   2321 kwargs.pop('ylog', True)
   2322 Plot.plot(self, self.x, self.y, yerr=self.yerr, xerr=self.xerr,
   2323           title=self.title, xlabel=self.xlabel, ylabel=self.ylabel,
   2324           overplot=overplot, clearwindow=clearwindow, **kwargs)
-> 2325 super().hline(y=0, xmin=0, xmax=1, linecolor='k',
   2326               linewidth=.8, overplot=True)

File /lagado.real/sherpa/sherpa-main/sherpa/plot/__init__.py:388, in Plot.hline(y, xmin, xmax, linecolor, linestyle, linewidth, overplot, clearwindow)
    383 @staticmethod
    384 def hline(y, xmin=0, xmax=1,
    385           linecolor=None, linestyle=None, linewidth=None,
    386           overplot=False, clearwindow=True):
    387     "Draw a line at constant y, extending over the plot."
--> 388     backend.hline(y, xmin=xmin, xmax=xmax, linecolor=linecolor,
    389                   linestyle=linestyle, linewidth=linewidth,
    390                   overplot=overplot, clearwindow=clearwindow)

File /lagado.real/sherpa/sherpa-main/sherpa/plot/backend_utils.py:82, in translate_args.<locals>.inner(self, *args, **kwargs)
     80             if val in transl:
     81                 kwargs[kw] = transl[val]
---> 82 return func(self, *args, **kwargs)

File /lagado.real/sherpa/sherpa-main/sherpa/plot/bokeh_backend.py:399, in BokehBackend.hline(self, y, xmin, xmax, linecolor, linestyle, linewidth, overplot, clearwindow)
    389 """Draw a horizontal line
    390 
    391 Parameters
   (...)
    395 {kwargs}
    396 """
    397 axes = self.setup_axes(overplot, clearwindow)
--> 399 line = Span(location=y, dimension='width',
    400             line_width=linewidth,
    401             line_color=linecolor,
    402             line_dash=linestyle)
    403 axes.add_layout(line)

File /lagado2.real/local/anaconda/envs/sherpa-main/lib/python3.11/site-packages/bokeh/models/annotations/geometry.py:432, in Span.__init__(self, *args, **kwargs)
    431 def __init__(self, *args, **kwargs) -> None:
--> 432     super().__init__(*args, **kwargs)

File /lagado2.real/local/anaconda/envs/sherpa-main/lib/python3.11/site-packages/bokeh/models/annotations/annotation.py:49, in Annotation.__init__(self, *args, **kwargs)
     48 def __init__(self, *args, **kwargs) -> None:
---> 49     super().__init__(*args, **kwargs)

File /lagado2.real/local/anaconda/envs/sherpa-main/lib/python3.11/site-packages/bokeh/models/renderers/renderer.py:131, in CompositeRenderer.__init__(self, *args, **kwargs)
    130 def __init__(self, *args: Any, **kwargs: Any) -> None:
--> 131     super().__init__(*args, **kwargs)

File /lagado2.real/local/anaconda/envs/sherpa-main/lib/python3.11/site-packages/bokeh/models/renderers/renderer.py:82, in Renderer.__init__(self, *args, **kwargs)
     81 def __init__(self, *args, **kwargs) -> None:
---> 82     super().__init__(*args, **kwargs)

File /lagado2.real/local/anaconda/envs/sherpa-main/lib/python3.11/site-packages/bokeh/model/model.py:119, in Model.__init__(self, *args, **kwargs)
    116 if "id" in kwargs:
    117     raise ValueError("initializing 'id' is not allowed")
--> 119 super().__init__(**kwargs)
    120 default_theme.apply_to_model(self)

File /lagado2.real/local/anaconda/envs/sherpa-main/lib/python3.11/site-packages/bokeh/core/has_props.py:306, in HasProps.__init__(self, **properties)
    304     if value is Undefined or value is Intrinsic:
    305         continue
--> 306     setattr(self, name, value)
    308 initialized = set(properties.keys())
    309 for name in self.properties(_with_props=True): # avoid set[] for deterministic behavior

File /lagado2.real/local/anaconda/envs/sherpa-main/lib/python3.11/site-packages/bokeh/core/has_props.py:338, in HasProps.__setattr__(self, name, value)
    336 properties = self.properties(_with_props=True)
    337 if name in properties:
--> 338     return super().__setattr__(name, value)
    340 descriptor = getattr(self.__class__, name, None)
    341 if isinstance(descriptor, property): # Python property

File /lagado2.real/local/anaconda/envs/sherpa-main/lib/python3.11/site-packages/bokeh/core/property/descriptors.py:331, in PropertyDescriptor.__set__(self, obj, value, setter)
    328     class_name = obj.__class__.__name__
    329     raise RuntimeError(f"{class_name}.{self.name} is a readonly property")
--> 331 value = self.property.prepare_value(obj, self.name, value)
    332 old = self._get(obj)
    333 self._set(obj, old, value, setter=setter)

File /lagado2.real/local/anaconda/envs/sherpa-main/lib/python3.11/site-packages/bokeh/core/property/bases.py:365, in Property.prepare_value(self, owner, name, value, hint)
    363 else:
    364     obj_repr = owner if isinstance(owner, HasProps) else owner.__name__
--> 365     raise ValueError(f"failed to validate {obj_repr}.{name}: {error}")
    367 if isinstance(owner, HasProps):
    368     obj = owner

ValueError: failed to validate Span(id='p1748', ...).line_color: expected either None or a value of type Color, got 'k'
```

This looks like it's because our code has some matlpotlib-isms in it - e.g. 

```
File /lagado.real/sherpa/sherpa-main/sherpa/plot/__init__.py:2325, in ResidPlot.plot(self, overplot, clearwindow, **kwargs)
   2321 kwargs.pop('ylog', True)
   2322 Plot.plot(self, self.x, self.y, yerr=self.yerr, xerr=self.xerr,
   2323           title=self.title, xlabel=self.xlabel, ylabel=self.ylabel,
   2324           overplot=overplot, clearwindow=clearwindow, **kwargs)
-> 2325 super().hline(y=0, xmin=0, xmax=1, linecolor='k',
   2326               linewidth=.8, overplot=True)
```

Line 2325 has the line color set to `k` and not some "generic" version.

I'm not sure, in our bright new world, where the logic for this should go (e.g. line 2325 or in the backend code or both or ...)

**Comment by hamogu** [comment]
That logic should already be in the bokeh backend, the lines https://github.com/sherpa/sherpa/blob/main/sherpa/plot/bokeh_backend.py#L383 has the `@translate_args` decorator, which should translate from 'k' to the generic form. The problem is that main does not have a translate layer for the `linecolor` but it should. Did I not just add that last week? Maybe it's not merged yet or was part of another of your PRs? I'll look, but right now I have to run and get Johann from afterschool.

**Comment by DougBurke** [comment]
There's no hurry here on my part.I just happened to be trying out bokeh (for #1679) and thought I'd try other plot styles.

**Comment by hamogu** [comment]
I's in https://github.com/sherpa/sherpa/pull/1976
If you build this on top of #1976 (or review that one so it gets merged and rebase) the 'k' problem will go away.

**Comment by DougBurke** [comment]
Fixed by #1976
